### PR TITLE
Add Impervious Table and associated tests

### PR DESCRIPTION
**Pull request by ELSW56**

### CODE REVIEW

File: preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy [hunk 1]
```diff
@@ -322,6 +322,43 @@ IProcess formatHydroLayer() {
...
+            outputTableName = "INPUT_IMPERVIOUS_${UUID.randomUUID().toString().replaceAll("-", "_")}"
+            datasource.execute """Drop table if exists $outputTableName;
+                        CREATE TABLE $outputTableName (THE_GEOM GEOMETRY(GEOMETRY, $epsg), id_impervious serial, ID_SOURCE VARCHAR);"""
+            logger.info(inputTableName)
+            if(inputTableName!=null){
+                datasource.withBatch(1000) { stmt ->
+                    datasource.getTable(inputTableName).eachRow { row ->
+                        def toAdd = true
+                        if ((row.surface != null) && (row.surface == "grass")) {toAdd = false}
+                        if ((row.parking != null) && (row.parking == "underground"))  {toAdd = false}
```
Comment (ebocher) [inline preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy]: What about is we use the same approach as getType to find the type associated to a set of tags.
Then, if the type is  null the values are not added.

File: preparedata/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayersTests.groovy [hunk 2]
```diff
@@ -50,5 +51,23 @@ class OSMGISLayersTests {
 
         //h2GIS.getTable(process.results.hydroTableName).save("./target/osm_hydro.shp")
         assertEquals 10, h2GIS.getTable(process.results.hydroTableName).rowCount
+
+        //h2GIS.getTable(process.results.imperviousTableName).save("./target/osm_hydro.shp")
+        assertEquals 45, h2GIS.getTable(process.results.imperviousTableName).rowCount
+    }
+
+    @Test
+    void createImperviousTableTest() {
```
Comment (ebocher) [inline preparedata/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayersTests.groovy]: merge this test with the creategislayers test

File: preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayers.groovy [hunk 3]
```diff
@@ -206,4 +222,46 @@ static Map readJSONParameters(def jsonFile) {
...
 
+/**
+ * This process is used to create the impervious table from an osm xml file
+ * @param datasource A connexion to a DB to load the OSM file
+ * @param osmFilePath The name of the OSM XML file
+ * @param epsg code to reproject the GIS layers, default is -1
+ *
+ * @return imperviousTableName The name of the resulting GIS table
+ */
+IProcess createImperviousTable() {
```
[the remark this answers is not in the document]
Comment (ELSW56) [reply preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayers.groovy]: It has been done. At first, I wanted a separate process to make specific tests. Should I remove it ?

File: preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy (hunk 1 above)
Comment (ELSW56) [reply preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy]: It's not exactly the same : there's no type to provide.
It's more like the special treatment for subways when formatting the rails layer.

File: preparedata/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayersTests.groovy (hunk 2 above)
Comment (ELSW56) [reply preparedata/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayersTests.groovy]: OK, I'll remove it.

File: preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayers.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (ebocher) [reply preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/OSMGISLayers.groovy]: Thanks

File: preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy (hunk 1 above)
Comment (ebocher) [reply preparedata/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/preparedata/osm/FormattingForAbstractModel.groovy]: ok

**Pull request merged**